configurations/dockerYamlTemplate.go

```diff
@@ -91,6 +93,23 @@ func convertMapToMap(mapInterface map[string]interface{}) map[string]string {
 	return mapString
 }
 
+// Parses the programmingLanguage instruction from yaml and returns an instance of `programmingLanguage`.
+func parseProgrammingLanguage(data map[string]interface{}) instruction {
+	convertedData := convertMapToMap(data)
+	var langObj programmingLanguage
+	if convertedData["name"] != "" {
+		langObj.Name = convertedData["name"]
+		if strings.Contains(langObj.Name, "+") {
```

Comment:
I'm guessing this is to search for g++?
We solve that issue by just not allowing g++ in the assignment-env.yaml and instead just allow gpp.

Reply:
Okay